pkl/private/pkl_codegen_java.bzl

```diff
@@ -114,6 +115,56 @@ _pkl_java_src_jar = rule(
     ],
 )
 
+def _pkl_java_resource_jar_impl(ctx):
+    srcjar = ctx.file.srcjar
+    resource_jar = ctx.outputs.out
+
+    ctx.actions.run_shell(
+        inputs = [srcjar],
+        outputs = [resource_jar],
+        command = """
+            set -e
+            EXECROOT="$PWD"
+            TMPDIR=$(mktemp -d)
+            trap "rm -rf $TMPDIR" EXIT
+
+            # Extract srcjar
+            unzip -q "$EXECROOT/$1" -d "$TMPDIR" || true
+
+            # Create resource JAR with resources/ prefix stripped
+            cd "$TMPDIR"
+            if [ -d "resources" ]; then
+                cd resources
+                zip -qr "$EXECROOT/$2" . || true
+            else
+                # Create empty JAR if no resources directory exists
+                touch empty.txt
+                zip -q "$EXECROOT/$2" empty.txt
+                rm empty.txt
```

Comment:
It looks like the following would generate an empty zip file; maybe ok as long as we keep the comment?

```
echo -ne '\x50\x4b\x05\x06\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' > "$EXECROOT/$2"
```